spring-batch-core/src/main/java/org/springframework/batch/core/JobParameters.java

```diff
@@ -269,8 +271,8 @@ public Properties toProperties() {
 		Properties props = new Properties();
 
 		for (Map.Entry<String, JobParameter> param : parameters.entrySet()) {
-			if(param.getValue() != null) {
-				props.put(param.getKey(), param.getValue().toString());
+			if (param.getValue() != null) {
+				props.put(param.getKey(), Optional.of(param.getValue()).map(Object::toString).orElse(""));
```

Comment:
I think this could be simplified to `Objects.toString(param.getValue(), "")`.

Reply:
Thanks for your comment. I've changed to `Objects.toString(param.getValue().toString(), ""))`.